src/main/java/de/bayern/gdi/gui/WMSMapSwing.java

```diff
@@ -407,10 +374,32 @@ public void onMouseDragged(MapMouseEvent ev) {
                 panel.add(
                         JMapStatusBar.createDefaultStatusBar(mapPane), "grow");
                 swingNode.setContent(panel);
+
+                setExtend(INITIAL_EXTEND_X1,
+                        INITIAL_EXTEND_X2,
+                        INITIAL_EXTEND_Y1,
+                        INITIAL_EXTEND_Y2,
+                        INITIAL_CRS);
             }
         });
     }
 
+    private void setExtend(Double x1, Double x2, Double y1, Double y2, String
+            crs) {
+        CoordinateReferenceSystem coordinateReferenceSystem = null;
+        try {
+            coordinateReferenceSystem = CRS.decode(crs);
+        } catch (FactoryException e) {
+            log.log(Level.SEVERE, e.getMessage(), e);
+        }
+        ReferencedEnvelope initExtend =
+                new ReferencedEnvelope(x1,
+                        x2,
+                        y1,
+                        y2, coordinateReferenceSystem);
+        MapViewport viewport = new MapViewport(initExtend, true);
```

Comment:
To zoom to a specific extent you dont need to create a new Viewport!
Take a look at http://docs.geotools.org/latest/javadocs/org/geotools/swing/AbstractMapPane.html#setDisplayArea-org.opengis.geometry.Envelope-